### ISSUE TITLE
Breaks figure environment?

### ISSUE THREAD

**Issue by jacobq** [agude/UMN-PhD-Thesis-Template#4]
I am using this (at least I think I am...I based a project on [the version on Overleaf that looks the same](https://www.overleaf.com/latex/templates/university-of-minnesota-thesis-template/tkngcptkrmgy)), and when I try to create a `figure` environment I encounter a run-away argument error that looks something like this:

> Runaway argument?
> 
> tbp]\end {figure} 
> ! Paragraph ended before \@ xfloat was complete.
> <to be read again> 
>                    \par 
> l.24 
>      
> I suspect you've forgotten a `}', causing me to apply this
> control sequence to too much text. How can we recover?
> My plan is to forget the whole thing and hope for the best.

At first I thought this might be due to my use of the [`setspace` package](https://www.ctan.org/pkg/setspace), but I can reproduce it without that:

```tex
%\documentclass{article} % <--- works when this is used instead
\documentclass[12pt]{mnthesis}

\begin{document}

% Nothing here, but it should still compile w/o issue
\begin{figure}
\end{figure}

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Elit ut aliquam purus sit amet luctus venenatis. Eu turpis egestas pretium aenean pharetra magna ac placerat.

\end{document}
```

I can't imagine that people are not putting figures in their thesis documents, so what's going on here? Maybe something with Overleaf? Maybe I am missing an option? Any help appreciated

**Comment by jacobq** [comment]
Seems like the simple example I showed above can be "fixed" by adding `[]` after `\begin{figure}`. However, this is not the case if `\usepackage[onehalfspacing]{setspace}` is used.

https://stackoverflow.com/questions/29174673/modified-xfloat-in-latex-causes-runaway-argument

**Comment by agude** [comment]
Hi @jacobq!

First, congrats on starting to write the thesis! 🥳 

Second, it looks like the source on Overleaf is the one that has been floating around the Minnesota servers for years and which I based this version off of (and, I hope, improved!). 

You can see that their version seems to match this commit from my repo: https://github.com/agude/UMN-PhD-Thesis-Template/commit/59dd3ec9a4b90a4fdc0678b97fb69afb1fe8946c#diff-f66bdfdf755804be1d77f736ec7e55b121d288bab7d49989b20a4aab4a5421ad

So I don't think it's a bug in my version. :-) Happy to help debug though when I have some free time this evening!

**Comment by jacobq** [comment]
Thanks for the response! When I used your version in my minimal test case the problem went away. However, when I tried using it on my actual thesis a number of things broke. For example, it looks like you removed the `campus` command (`\def\campus#1{\gdef\@campus{#1}}`).  Is there a changelog somewhere that I can reference for figuring out what I need to change?

**Comment by agude** [comment]
I don't have a change log, except for git where I try to write meaningful commit statements.

This is the "first" commit of the thesis, which probably matches what is on Overleaf: https://github.com/agude/UMN-PhD-Thesis-Template/commit/59dd3ec9a4b90a4fdc0678b97fb69afb1fe8946c

From there you can walk through history and look at the diffs.

This is the "last meaningful" commit: https://github.com/agude/UMN-PhD-Thesis-Template/commit/ed030b768215b267da158786c9947eee4b611ffa

So not a ton to search through!

**Comment by agude** [comment]
@jacobq:

Here is where I stripped out commands that I thought were redundant: https://github.com/agude/UMN-PhD-Thesis-Template/commit/ed030b768215b267da158786c9947eee4b611ffa

If I accidentally killed something important/useful, please let me know! 😄

**Comment by jacobq** [comment]
Thanks for your help! For now I've just pasted those commands back in my copy to make it compile OK (didn't feel up to starting from scratch and copying everything over). It probably wouldn't have been too hard to do your way, and I don't have anything against your simplifications, but once I had things working I didn't want to mess with it any more :) (somehow I made it to the end of an MS CSci degree without ever mastering TeX, though I can usually StackExchange my way along).
